src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -422,12 +422,30 @@ internal partial class CultureData
             {
                 return CultureData.Invariant;
             }
-
-            // First check if GetCultureData() can find it (ie: its a real culture)
-            CultureData? retVal = GetCultureData(cultureName, useUserOverride);
-            if (retVal != null && !retVal.IsNeutralCulture)
+            CultureData? retVal = null;
+            try
             {
-                return retVal;
+                // First check if GetCultureData() can find it (ie: its a real culture)
+                retVal = GetCultureData(cultureName, useUserOverride);
+                if (retVal != null && !retVal.IsNeutralCulture)
+                {
+                    return retVal;
+                }
+            }
+            catch (CultureNotFoundException) when (GlobalizationMode.PredefinedCulturesOnly)
```

Comment:
>CultureNotFoundException [](start = 19, length = 24)

why we are catching this exception here? I am seeing GetCultureData is not throwing exceptions

Reply:
`GetCultureData` doesn't throw exceptions but `CreateCultureData` (L720) does

Reply:
but we are not calling CreateCultureData  inside the try block. am I missing something? I mean the catch block will never execute. no?

---
In reply to: [586701928](https://github.com/dotnet/runtime/pull/47301#discussion_r586701928) [](ancestors = 586701928)

Reply:
CreateCultureData is called in GetCultureData

Reply:
Got it. `CreateCultureData` shouldn't be allowed to throw. this was the original design. I am seeing it is now because you have started calling `NlsIsEnsurePredefinedLocaleName` and `IcuIsEnsurePredefinedLocaleName` methods which throw.

I suggest tweaking the `NlsIsEnsurePredefinedLocaleName` and `IcuIsEnsurePredefinedLocaleName` to take a parameter telling to throw or not and then return bool if succeeded or failed. and let CreateCultureData check this `bool` and not throw.
Or let `NlsIsEnsurePredefinedLocaleName` and `IcuIsEnsurePredefinedLocaleName` just return bool and then throw at the places calling it when it makes sense to throw.

Reply:
I see. Could I move the throw to `GetCultureData` then?

Reply:
Let GetCultureData don't throw and instead make it return null when there is failures. the callers of GetCultureData usually throw the exceptions. you may look at the following as example:

https://github.com/dotnet/runtime/blob/ba8ce9e1a00b57aba6ab7384c16f4594be6754e2/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs#L173

Reply:
ok got it. that makes sense. i will change that then